mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixture.scala

```diff
@@ -121,6 +124,17 @@ class GaussianMixture private (
    */
   def getConvergenceTol: Double = convergenceTol
 
+  /**
+   * Sets indicator for computing Gaussians on driver or executors
+   */
+  def setDistributeGaussians(distributeGaussians: Boolean): this.type = {
```

Comment:
I'm not sure whether we should expose this flag. It should be easy to calculate the computation cost and communication cost and decide whether we want to distribute the computation. A simple heuristic using a rough estimate should work well in practice.